beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -6455,13 +6455,18 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         &self,
         block_root: &Hash256,
     ) -> Result<Option<(LightClientBootstrap<T::EthSpec>, ForkName)>, Error> {
-        let Some((state_root, slot)) = self
-            .get_blinded_block(block_root)?
-            .map(|block| (block.state_root(), block.slot()))
-        else {
+        let runtime =  tokio::runtime::Runtime::new().map_err(|_| Error::RuntimeShutdown)?;
+
+        let Some(block) = runtime.block_on( async {
+            self
+                .get_block(block_root).await
+            }
+        )? else {
```

Comment:
Note that .get_block() fetches an execution payload from the EL, which may timeout blocking this task for maybe seconds? get_light_client_bootstrap is called by API and ReqResp:
- API spawns a blocking task, so it's ok to block

https://github.com/sigp/lighthouse/blob/31044402ee180ff937027ec842513bef90d7eec8/beacon_node/http_api/src/lib.rs#L2421-L2422

- ReqResp spawns a blocking fn, so ok to block too

https://github.com/sigp/lighthouse/blob/441fc1691b69f9edc4bbdc6665f3efab16265c9b/beacon_node/beacon_processor/src/lib.rs#L1413-L1414

Having to wait for the EL should only degrade the experience for the caller. This is ok since this penalty already exist for queries on the API and ReqResp that involves full blocks